lib/modem_slm/CMakeLists.txt

```diff
@@ -0,0 +1,11 @@
+#
+# Copyright (c) 2023 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+zephyr_library()
+zephyr_library_sources_ifdef(CONFIG_MODEM_SLM modem_slm.c)
+zephyr_library_sources_ifdef(CONFIG_MODEM_SLM modem_slm_monitor.c)
```

Comment:
`_ifdef` is redundant here, the file won't be included if `CONFIG_MODEM_SLM` is disabled. Just define sources unconditionally.